builtin/providers/aws/resource_aws_sns_topic_subscription.go

```diff
@@ -192,6 +218,76 @@ func subscribeToSNSTopic(d *schema.ResourceData, snsconn *sns.SNS) (output *sns.
 		return nil, fmt.Errorf("Error creating SNS topic: %s", err)
 	}
 
-	log.Printf("[DEBUG] Created new subscription!")
+	log.Printf("[DEBUG] Finished subscribing to topic %s with subscription arn %s", topic_arn, *output.SubscriptionArn)
+
+	if strings.Contains(protocol, "http") && subscriptionHasPendingConfirmation(output.SubscriptionArn) {
+
+		log.Printf("[DEBUG] SNS create topic subscritpion is pending so fetching the subscription list for topic : %s (%s) @ '%s'", endpoint, protocol, topic_arn)
+
+		for i := 0; i < max_fetch_retries && subscriptionHasPendingConfirmation(output.SubscriptionArn); i++ {
+
+			subscription, err := findSubscriptionByNonID(d, snsconn)
+
+			if err != nil {
+				return nil, fmt.Errorf("Error fetching subscriptions for SNS topic %s: %s", topic_arn, err)
+			}
+
+			if subscription != nil {
+				output.SubscriptionArn = subscription.SubscriptionArn
+				break
+			}
+
+			time.Sleep(time.Second * fetch_retry_delay)
+		}
+
+		if subscriptionHasPendingConfirmation(output.SubscriptionArn) {
+			return nil, fmt.Errorf("Endpoint (%s) did not autoconfirm the subscription for topic %s", endpoint, topic_arn)
+		}
+	}
+
+	log.Printf("[DEBUG] Created new subscription! %s", *output.SubscriptionArn)
 	return output, nil
 }
+
+// finds a subscription using protocol, endpoint and topic_arn (which is a key in sns subscription)
+func findSubscriptionByNonID(d *schema.ResourceData, snsconn *sns.SNS) (*sns.Subscription, error) {
+	protocol := d.Get("protocol").(string)
+	endpoint := d.Get("endpoint").(string)
+	topic_arn := d.Get("topic_arn").(string)
+
+	req := &sns.ListSubscriptionsByTopicInput{
+		TopicArn: aws.String(topic_arn),
+	}
+
+	for {
+
+		res, err := snsconn.ListSubscriptionsByTopic(req)
+
+		if err != nil {
+			return nil, fmt.Errorf("Error fetching subscripitions for topic %s : %s", topic_arn, err)
+		}
+
+		for _, subscription := range res.Subscriptions {
+			log.Printf("[DEBUG] check subscription with EndPoint %s, Protocol %s,  topicARN %s and SubscriptionARN %s", *subscription.Endpoint, *subscription.Protocol, *subscription.TopicArn, *subscription.SubscriptionArn)
+			if *subscription.Endpoint == endpoint && *subscription.Protocol == protocol && *subscription.TopicArn == topic_arn && !subscriptionHasPendingConfirmation(subscription.SubscriptionArn) {
+				return subscription, nil
+			}
+		}
+
+		// if there are more than 100 subscriptions then go to the next 100 otherwise return nil
+		if res.NextToken != nil {
+			req.NextToken = res.NextToken
+		} else {
+			return nil, nil
+		}
+	}
+}
+
+// returns true if arn is nil or has both pending and confirmation words in the arn
+func subscriptionHasPendingConfirmation(arn *string) bool {
+	if arn != nil && !strings.Contains(strings.ToLower(*arn), "pending") && !strings.Contains(strings.ToLower(*arn), "confirmation") {
```

Comment:
Is there any reason why not just do this? Have you spotted any edge cases where the casing was different?

```go
if arn != nil && *arn != "Pending confirmation" {
```